crates/database/interface/src/lib.rs

```diff
@@ -21,7 +21,7 @@ pub use empty_db::{EmptyDB, EmptyDBTyped};
 #[auto_impl(&mut, Box)]
 pub trait Database {
     /// The database error type.
-    type Error;
+    type Error: core::error::Error;
```

Comment:
I would consider adding a `'static` bound here, as described in https://github.com/bluealloy/revm/pull/1840/files#r1819000225